test/test_power_flow_in_the_loop.jl

```diff
@@ -54,6 +54,108 @@
+                    r = real(split_impedance),
+                    x = imag(split_impedance),
+                    b = split_shunt,
+                    angle_limits = get_angle_limits(line),
+                    rating = get_rating(line),
+                )
+                i == 1 && display(l)
+                add_component!(system, l)
+            end
+        end
+        template = get_template_dispatch_with_network(
+            NetworkModel(
+                PTDFPowerModel;
+                PTDF_matrix = PTDF(system),
+                power_flow_evaluation = ACPowerFlow(),
+            ),
+        )
+        model_m = DecisionModel(template, system; optimizer = HiGHS_optimizer)
+        @test build!(model_m; output_dir = mktempdir(; cleanup = true)) ==
+              PSI.ModelBuildStatus.BUILT
+
+        @test solve!(model_m) == PSI.RunStatus.SUCCESSFULLY_FINALIZED
+        results = OptimizationProblemResults(model_m)
+        vd = read_aux_variables(results)
+        display(vd["PowerFlowLineActivePowerFromTo__Line"])
+        if replace_line
+            name = "$(get_name(line))_1"
+            parallel_line_flow =
+                vd["PowerFlowLineActivePowerFromTo__Line"][:, name][1] +
+                im * vd["PowerFlowLineReactivePowerFromTo__Line"][:, name][1]
+        else
+            name = get_name(line)
+            original_line_flow =
+                vd["PowerFlowLineActivePowerFromTo__Line"][:, name][1] +
+                im * vd["PowerFlowLineReactivePowerFromTo__Line"][:, name][1]
+        end
+    end
+
+    @test isapprox(
+        2 * parallel_line_flow,
+        original_line_flow,
+        atol = 1e-3,
+    )
+end
+
+@testset "network reduction error" begin
+    system = build_system(PSITestSystems, "c_sys5_uc")
+    # replace a line with a breaker-switch, so that we get a network reduction.
+    line = get_component(Line, system, "1")
+    remove_component!(system, line)
+    bs = PSY.DiscreteControlledACBranch(
+        ;
+        name = get_name(line),
+        available = true,
+        active_power_flow = 0.0,
+        reactive_power_flow = 0.0,
+        arc = get_arc(line),
+        r = 0.0,
+        x = 0.0,
+        rating = get_rating(line),
+        discrete_branch_type = PSY.DiscreteControlledBranchType.BREAKER,
+        branch_status = PSY.DiscreteControlledBranchStatus.CLOSED,
+    )
+    add_component!(system, bs)
+    template = get_template_dispatch_with_network(
+        NetworkModel(
+            PTDFPowerModel;
+            PTDF_matrix = PTDF(system),
+            power_flow_evaluation = ACPowerFlow(),
+        ),
+    )
+    model_m = DecisionModel(template, system; optimizer = HiGHS_optimizer)
+    # ideally, would @test_logs for IS.NotImplementedError, but build! catches all errors
+    # and does `@error "DecisionModel Build Failed"`, attaching the original error as
+    # a field named "exception." This means @test_logs can't see the original error message.
```

Comment:
You can use `PSI.build_impl!` to achieve `@test_logs`, see https://github.com/NREL-Sienna/PowerSimulations.jl/blob/9ec20bc98bac0005637686d07217c869b2b2f902/test/test_market_bid_cost.jl#L934-L939